sql-plugins/src/main/java/io/cdap/pipeline/sql/plugins/transforms/FilterSQLTransform.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.pipeline.sql.plugins.transforms;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.pipeline.sql.api.template.SQLNode;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.sql.SqlBinaryOperator;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.tools.RelBuilder;
+
+import javax.annotation.Nullable;
+
+/**
+ * A SQL Projection transform.
+ */
+@Plugin(type = SQLNode.PLUGIN_TYPE)
+@Name("Filter")
+@Description("The Projection transform lets you add a filter to an SQL query.")
+public class FilterSQLTransform extends SQLNode {
+  private static final String LEFT_VALUE_DESC = "The value which forms the left-hand operand of the comparison.";
+  private static final String LEFT_TYPE_DESC = "The type of the left-hand operand.";
+  private static final String RIGHT_VALUE_DESC = "The value which forms the right-hand operand of the comparison.";
+  private static final String RIGHT_TYPE_DESC = "The type of the right-hand operand.";
+  private static final String OPERATION_DESC = "The operation for this filter.";
+
+  /**
+   * Config class for ProjectionTransform
+   */
+  public static class FilterTransformTransformConfig extends PluginConfig {
+    @Description(LEFT_VALUE_DESC)
```

Comment:
this is kind of verbose and only allows certain types of filters. I think it's fine for demo purposes, but ideally we would want a user to also be able to specify sql. Have we looked at how to parse sql into relational algebra using Calcite (don't need to do this now).

Reply:
I believe it only supports specific dialects, but it is possible to parse SQL. However, in those cases I'm not entirely sure whether we'd be able to inline those queries automatically.